kroxylicious-filters/kroxylicious-record-encryption/src/test/java/io/kroxylicious/filter/encryption/RecordEncryptionFilterTest.java

```diff
@@ -169,6 +187,33 @@ void shouldNotEncryptTopicWithoutKeyId() {
         verify(encryptionManager, never()).encrypt(any(), anyInt(), any(), any(), any());
     }
 
+    @Test
+    void shouldCountPlainAndEncryptedRecords() {
+        // Given
+        var produceRequestData = buildProduceRequestData(new TopicProduceData()
+                .setName(ENCRYPTED_TOPIC)
+                .setPartitionData(List.of(new PartitionProduceData().setRecords(makeRecord(HELLO_CIPHER_WORLD)))),
+                new TopicProduceData()
+                        .setName(UNRESOLVED_TOPIC)
+                        .setPartitionData(List.of(new PartitionProduceData().setRecords(makeRecord(HELLO_PLAIN_WORLD)))));
+
+        // When
+        encryptionFilter.onProduceRequest(ProduceRequestData.HIGHEST_SUPPORTED_VERSION, new RequestHeaderData(), produceRequestData, context);
+
+        // Then
+        assertThat(Metrics.globalRegistry.get("kroxylicious_filter_record_encryption_plain_records").counter())
+                .isNotNull()
+                .satisfies(counter -> assertThat(counter.getId()).isNotNull())
+                .satisfies(counter -> assertThat(counter.count())
+                        .isGreaterThan(0));
```

Comment:
As this is a unit test we can be explicit about what to expect so we can do 
```suggestion
                        .isEqualTo(1.0));
```
Because we know its operating on exactly the records we expect and there is no background processing.